docs/source/learn-prototype.md

```diff
@@ -0,0 +1,109 @@
+GridSearch, Cross Validation, Pipeline
+======================================
+
+
+Motivation
+----------
+
+Scikit learn already parallelizes through Joblib.
+What are their current pain points?
+
+1.  Grid searches of pipelines don't reuse shared results.
+    For example, consider the following search:
+
+    pipe = Pipeline(A->B->C)
+    search = ParameterSearch(pipe, [(a1, b1, c1), (a1, b1, c2), ..., (ai, bj, ck)])
+
+    Currently scikit-learn evaluates A with parameters a1 many times without
+    reuse.  By better tracking intermediate results we can avoid many repeated
+    computations
+2.  Nesting of parallelism.  When two systems try to parallelize at once they
+    interfere with each other.  (does this happen?)
+3.  Out-of-core cross validation with partial fit
+4.  Proper distributed computing
+
+
+Objective
+---------
+
+We'd like a tiny proof of concept library that includes `Pipeline` and
+`ParameterSearch`.  These don't need to support the full API but do need to be
+useful on a common case problem.
+
+Our model problem is to create a Pipeline of scikit learn estimators:
+
+    pipe = Pipeline(A, B, C)
+
+And then search over a parameter space within this pipeline:
+
+    grid = GridSearchCV(pipe, {parameter space})
+
+
+Plan
+----
+
+We create the following class hierarchy
+
+    Base - DaskEstimator - ParameterSearch - GridSearch
+                         \
+                          - Pipeline
+
+DaskEstimator supports the following subset of the scikit-learn API
+
+    clone
+    fit
```

Comment:
It's likely that fit can defer any actual computation until predict or score is called.